### PR TITLE
Updated for new torchtext and solved issue with 100% accuracy

### LINKED ISSUE #135: Issue with chapter 15 Sentiment Analysis with LSTM. Getting 100% accuracy while training.
Hello @rasbt. 
I am facing some issue with chapter 15 part 2 notebook. I tried running the notebook on colab.
Issues:
1. According to the book the IMDB test data should have 25k samples. Currently the IMDB test data has 12.5k samples.
2. Need for installation of portalocker.
3. While training I am getting 100% validation accuracy right from from first epoch. (100% accuracy for all further epochs), and the test accuracy is also 100%.
Screenshot 2023-07-15 175804

### PR DESCRIPTION
**Pull request by sabre-code**
Found the issue with 100% accuracy. The labels in IMDB dataset were "pos" and "neg" earlier. Now they are 1 and 2 for positive and negative respectively. Tested on Colab and got expected results.
Screenshot 2023-07-15 175804

fixes #135

### CODE REVIEW

**Comment by rasbt** [comment]
Thanks for the patch. I just checked and it works for me. I made some additional changes that will also make it work for readers who use the older torchtext version (it's just an if/else version check). Somehow I can't push to this branch due to a permission error (maybe "allowing contributions from maintainers" was disabled for this PR) so I will do that after merging. Thanks for the fix!

**Comment by sabre-code** [comment]
Sounds great. Thanks a lot for this great book. ♥️
